applications/asset_tracker_v2/src/cloud/cloud_codec/cloud_codec.h

```diff
@@ -319,6 +320,21 @@ int cloud_codec_encode_cloud_location(
 	struct cloud_codec_data *output,
 	struct cloud_data_cloud_location *cloud_location);
 
+/**
+ * @brief Decode received cloud location data.
+ *
+ * @param[in] input String buffer with encoded cloud location data.
+ * @param[in] input_len Length of input.
+ * @param[out] location Where to store the decoded cloud location data.
+ *
+ * @retval 0 on success.
+ * @retval -EFAULT if cloud returned with error to the location request.
+ * @retval -ENOMEM if codec couldn't allocate memory.
+ * @retval -ENOTSUP if the function is not supported by the backend.
```

Comment:
Maybe I'm nitpicking, but I'm still wondering about the return values. The relevant ones for the implementation are `-EFAULT` and `-ENOTSUP`, because these are handled differently in `cloud_location_data_handle()`. These are documented in the API, so no problems there. However, the nRF Cloud implementation can also return at least `-EBADMSG`, which is not documented here.

The returned error values can be somewhat implementation specific, so maybe we could only list the relevant `-EFAULT` and `-NOTSUP` here, and allow other negative errnos to be returned as needed by the implementation. The error value is printed by `cloud_location_data_handle()`, so it may add value to return it as it is, instead of converting the error to a more generic value.

The same may be true for multiple `cloud_codec_` functions, but I'm not asking you to change everything. 😄

Reply:
I'll add @return statement there:
`@return 0 on success or negative error value on failure.`

There's also a separate `@retval 0` line but I don't want to break the consistency within the file with this `@return` so leaving that also.